src/vmm/src/lib.rs

```diff
@@ -303,6 +329,42 @@ impl VMM {
         Ok(())
     }
 
+    fn add_block_device(
```

Comment:
This can also be unit tested and we can check for side effects as follows:
```rust
    #[test]
    fn test_add_block() {
        let vmm_config = default_vmm_config();
        let mut vmm = mock_vmm(vmm_config);

        let tempfile = TempFile::new().unwrap();
        let block_config = BlockConfig {
            path: tempfile.as_path().to_path_buf()
        };
        let mut cmdline = cmdline::Cmdline::new(4096);
        assert!(vmm.add_block_device(&block_config, &mut cmdline).is_ok());
        assert_eq!(vmm.block_devices.len(), 1);
        assert!(cmdline.as_str().contains("virtio"));

        let invalid_block_config = BlockConfig {
            // Let's create the tempfile directly here so that it gets out of scope immediately
            // and delete the underlying file.
            path: TempFile::new().unwrap().as_path().to_path_buf()
        };
        let mut cmdline = cmdline::Cmdline::new(4096);
        let err = vmm.add_block_device(&invalid_block_config, &mut cmdline).unwrap_err();
        assert!(matches!(err, Error::Block(block::Error::OpenFile(io_err)) if io_err.kind() == ErrorKind::NotFound));

        // The current implementation of the VMM does not allow more than one block device
        // as we are hard coding the `MmioConfig`.
        let mut cmdline = cmdline::Cmdline::new(4096);
        assert!(vmm.add_block_device(&block_config, &mut cmdline).is_err());
    }
```